python/cudf/cudf/core/column/column.py

```diff
@@ -1175,6 +1175,9 @@ def as_column(arbitrary, nan_as_null=None, dtype=None, length=None):
 
     elif isinstance(arbitrary, np.ndarray):
```

Comment:
Can we just refactor this whole block to work via `__array_interface__` instead of explicitly checking for `np.ndarray`?

And then maybe move it below the `isinstance(arbitrary, pa.Array)` block.

Reply:
Refactored, ready for review.